hack/delete-cluster.sh

```diff
@@ -6,29 +6,36 @@ set -o pipefail
 
 function usage() {
   echo "This script delete a kube cluster by kind."
-  echo "Usage: hack/delete-cluster.sh <CLUSTER_NAME> <KUBECONFIG>"
-  echo "Example: hack/delete-cluster.sh host /root/.kube/karmada.config"
+  echo "Usage: hack/delete-cluster.sh <CLUSTER_NAME> <KUBECONFIG> or hack/delete-cluster.sh --all"
+  echo "Example: hack/delete-cluster.sh karmada-host /root/.kube/karmada.config"
 }
 
-if [[ $# -ne 2 ]]; then
-  usage
-  exit 1
-fi
+REPO_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+source "${REPO_ROOT}"/hack/util.sh
 
-CLUSTER_NAME=$1
-if [[ -z "${CLUSTER_NAME}" ]]; then
-  usage
-  exit 1
-fi
-KUBECONFIG=$2
-if [[ -z "${KUBECONFIG}" ]]; then
-  usage
-  exit 1
-fi
+# befor delete cluster, check kind command exist, Ensure that it can be deleted directly after renaming
+util::cmd_must_exist "kind"
 
-# The context name has been changed when creating clusters by 'create-cluster.sh'.
-# This will result in the context can't be removed by kind when deleting a cluster.
-# So, we need to change context name back and let kind take care about it.
-kubectl config rename-context "${CLUSTER_NAME}" "kind-${CLUSTER_NAME}" --kubeconfig="${KUBECONFIG}"
+case $1 in
+    all)
+        kind delete clusters --all
```

Comment:
That's great to provide the `all` parameter, I think lots of people will use it.
But I am a little worried about deleting all clusters if I have other 'kind clusters' unrelated to Karmada, those will delete too. Shall we just delete the clusters that we will create?

Reply:
BTW, the others changes seem all right.

Reply:
I filtered by name，Maybe there's something better.